spark/v3.1/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/MergeIntoExec.scala

```diff
@@ -87,19 +89,21 @@ case class MergeIntoExec(
     val notMatchedPreds = params.notMatchedConditions.map(generatePredicate(_, joinedAttrs))
     val notMatchedProjs = params.notMatchedOutputs.map(_.map(generateProjection(_, joinedAttrs)))
     val projectTargetCols = generateProjection(params.targetOutput, joinedAttrs)
-    val nonMatchedPairs =   notMatchedPreds zip notMatchedProjs
+    val nonMatchedPairs = notMatchedPreds zip notMatchedProjs
     val matchedPairs = matchedPreds zip matchedProjs
 
     /**
      * This method is responsible for processing a input row to emit the resultant row with an
-     * additional column that indicates whether the row is going to be included in the final
-     * output of merge or not.
-     * 1. If there is a target row for which there is no corresponding source row (join condition not met)
-     *    - Only project the target columns with deleted flag set to false.
-     * 2. If there is a source row for which there is no corresponding target row (join condition not met)
-     *    - Apply the not matched actions (i.e INSERT actions) if non match conditions are met.
-     * 3. If there is a source row for which there is a corresponding target row (join condition met)
-     *    - Apply the matched actions (i.e DELETE or UPDATE actions) if match conditions are met.
+     * additional column that indicates whether the row is going to be included in the final output
+     * of merge or not.
+     *   1. If there is a target row for which there is no corresponding source row (join condition
```

Comment:
this shouldn't be randomly joined as well

Reply:
Fixed